### ISSUE TITLE
Clarify model relationships

### ISSUE THREAD

**Issue by quinnleong** [zjernst/soundscape#5]
I'm a little confused about the relationships between tracks, environments, collections, tags, and taggings. Are the tags referring to environments? Or are they a totally separate set? What about collections?

**Comment by zjernst** [comment]
The way I'm currently thinking of it:

Environments have many tracks (tracks belong to environment)
Tracks have many tags, tags have many tracks (taggings is the join table)

So tags are associated with Tracks and not environments. Basically, I want users to be able to give tracks tags like "calm", "peaceful", "moving", "warm", "busy", etc. that allow better searches (a search on "calm" might pull some from Ocean and Wind environments, but not the "stormy" tracks for instance).

Collections belong to a user
Collections have many tracks

Collections are a way to tie tracks to users (simulating purchasing, or letting them add them to their userpage). They don't have any functionality besides being a group that a user can customize.

Additionally, I further complicated it by adding Playlists (which I realized I left off). They're similar to collection (belong to user, have many tracks). Maybe Collection could incapsulate Playlists though, and just let a User have many Collections (and let "playlists" be a subset of all of them). 

I'm totally open to any and all feedback regarding this! I'm not sure what the conventional way to handle playlists/collections is.

**Comment by quinnleong** [comment]
That's super helpful, thanks!

Regarding collections and playlists, is the idea that users could have any tracks from the site in their playlists, but only tracks they have "bought" get added to their collection? If so, I think realistically it would be best to choose just one to flesh out. My suggestion would be playlists, since handling secure monetary transactions is beyond the scope of the class. And then if you wanted, you could also make a collection and have it simply be the collection of all tracks a user has added to any of their playlists; that wouldn't take too much extra work.

**Comment by zjernst** [comment]
Yeah, that was similar to the idea. I was thinking a user could add any tracks to their collection (to have a validation later if I got to 'buying') and could separately put songs in a playlist. So collection is a store of music a user liked and wanted to keep track of, and a playlist was specifically intended to be played in sequence (such that not every track in a collection would make it into a playlist, and that multiple playlists could all contain the same track if desired).

To illustrate:
User1:
Collection: [stormy_night, rain, rain_hard, lightning, wind, crickets, fireplace, birds]
Playlist "Night": [stormy_night, crickets, fireplace]
Playlist "Weather": [stormy_night, rain, rain_hard]

But I think you're right and I should just pick one to focus on, maybe move the other to bonus. Playlist seems more important, and could just be a requirement to add to a 'collection' anyway (i.e. your idea in collection is just tracks that have been added to a playlist)

Thanks so much for your feedback!

(**bonus bonus** my economy wasn't going to be secure money transaction, but you 'collect' a track by spending a point, and every time you upload a track you get 2 points or something similar. But that's very much not vital and can be revisited much later down the line. I might flesh out collection as part of this bonus phase if I get there)
